### PR TITLE
fix: 🐛 minor UI changes, ent flag for worker filter

### PR DESCRIPTION
**Pull request by DhariniJeeva**
# Description
this PR fixes a few changes - 
1. Introduce a flag to hide/show worker filter (we only show this field in enterprise)
2. disable and add helper text for disable_cred_rotation when dynamic cred is selected
3. role_Arn text update

## Screenshots (if appropriate)
 
Screenshot 2024-10-10 at 11 44 36 AM

Screenshot 2024-10-10 at 11 44 32 AM

## Checklist

- [x] I have added before and after screenshots for UI changes
- ~[ ] I have added JSON response output for API changes~
- ~ [ ] I have added steps to reproduce and test for bug fixes in the description
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- ~[ ] I have added tests that prove my fix is effective or that my feature works~

### CODE REVIEW

**Review by cameronperera — APPROVED** [review]
LGTM

**Comment by madeleinetle** [comment]
There's a large amount of spacing here. Is that noted to be fixed later? Also, the spacing is bit larger than expected on the radio button group. Otherwise, everything else looks good!
Screenshot 2024-10-10 at 11 48 51 AM

Screenshot 2024-10-10 at 11 47 38 AM

**Review by madeleinetle — COMMENTED** [review]
https://github.com/hashicorp/boundary-ui/pull/2517#issuecomment-2405817373

Left comments here

File: ui/admin/app/components/form/host-catalog/aws/index.hbs [hunk 1]
```diff
@@ -364,46 +364,63 @@
...
-    checked={{@model.disable_credential_rotation}}
-    disabled={{form.disabled}}
-    {{on 'change' (fn this.toggleDisableCredentialRotation @model)}}
-    as |F|
-  >
-    <F.Label>
-      {{t 'form.disable_credential_rotation.label'}}
-    </F.Label>
-  </Hds::Form::Checkbox::Field>
+  {{#if (feature-flag 'dynamic-credentials-worker-filter')}}
```
Comment (cameronperera) [inline ui/admin/app/components/form/host-catalog/aws/index.hbs:367]: Can we add a tests to verify this is hidden and displayed when the feature is false and true?

Comment (cameronperera) [inline ui/admin/app/components/form/host-catalog/aws/index.hbs:367]: Can we add a tests to verify this is hidden and displayed when the feature is false and true?

**Review by cameronperera — CHANGES_REQUESTED** [review]

File: addons/core/translations/resources/en-us.yaml [hunk 2]
```diff
@@ -212,6 +212,8 @@ host-catalog:
   help:
     static: Provide a list of hosts and host sets with static addresses
     plugin: Automate discovery of target hosts and services on Azure or AWS
+    role_arn: The ARN (Amazon Resource Name) used by this instance.
+    disable_credential_rotation: Credential rotation is automatically disabled when you use dynamic credentials
```
Comment (lisbet-alvarez) [inline addons/core/translations/resources/en-us.yaml:216]: small nitpick
```suggestion
    disable_credential_rotation: Credential rotation is automatically disabled when you use dynamic credentials.
```

File: ui/admin/config/features.js [hunk 3]
```diff
@@ -28,6 +28,7 @@ const baseEdition = {
   'target-network-address': false,
   'vault-worker-filter': false,
   'ldap-auth-methods': false,
+  'dynamic-credentials-worker-filter': false,
```
[the remark this answers is not in the document]
Comment (DhariniJeeva) [reply ui/admin/config/features.js:31]: my 2 cents: we should probably consolidate them, for example, vault-worker-filter is used the same way as this one. we should prob make a generic worker filter flag that can be shared.

File: ui/admin/app/components/form/host-catalog/aws/index.hbs (hunk 1 above)
Comment (DhariniJeeva) [reply ui/admin/app/components/form/host-catalog/aws/index.hbs:372]: added

File: ui/admin/app/components/form/host-catalog/aws/index.hbs [hunk 4]
```diff
@@ -217,7 +217,7 @@
         {{t 'resources.storage-bucket.form.role_arn.label'}}
       </F.Label>
       <F.HelperText>
-        {{t 'resources.storage-bucket.form.role_arn.help'}}
```
[the remark this answers is not in the document]
Comment (DhariniJeeva) [reply ui/admin/app/components/form/host-catalog/aws/index.hbs:220]: yeah good call , fixed

**Comment by DhariniJeeva** [comment]
> There's a large amount of spacing here. Is that noted to be fixed later? Also, the spacing is bit larger than expected on the radio button group. Otherwise, everything else looks good! Screenshot 2024-10-10 at 11 48 51 AM
> 
> Screenshot 2024-10-10 at 11 47 38 AM

I fixed the spacing for radio fields. The one is a shared component, so I'll add a ticket to the backlog. We prob want to fix the space in other places where we use this component, so it makes sense to fix it at the component level.

File: ui/admin/app/components/form/host-catalog/aws/index.hbs [hunk 5]
```diff
@@ -364,46 +364,63 @@
...
+        </F.Error>
+      {{/if}}
+    </Hds::Form::Textarea::Field>
+  {{/if}}
 
+  {{#if this.showDynamicCredentials}}
+    <Hds::Form::Checkbox::Field
+      name='disable_credential_rotation'
+      checked={{true}}
+      disabled={{true}}
```
[the remark this answers is not in the document]
Comment (DhariniJeeva) [reply ui/admin/app/components/form/host-catalog/aws/index.hbs:407]: we can't use readonly for checkbox fields

**Review by madeleinetle — APPROVED** [review]

**Review by ZedLi — APPROVED** [review]

**Pull request merged**